shm/Cargo.hpp

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <string>
+
+class Cargo {
+protected:
```

Comment:
Why protected? I don't see any derived class which will inherit from Cargo and use these values.